collect_app/src/main/java/org/odk/collect/android/upload/InstanceGoogleSheetsUploader.java

```diff
@@ -362,21 +364,19 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
         } else if (hasRepeatableGroups(element)) {
             answers.put(KEY, key);
         }
-        return makeAnswersFormattingResistant(answers);
+        return answers;
     }
 
-    public static HashMap<String, String> makeAnswersFormattingResistant(HashMap<String, String> answers) {
-        HashMap<String, String> fixedAnswers = new HashMap<>();
-        for (Map.Entry<String, String> item : answers.entrySet()) {
-            String value = item.getValue();
-            if (!value.startsWith("http") && !value.startsWith("=HYPERLINK")) {
-                // Avoid formatting answers https://stackoverflow.com/a/37827066/5479029
-                value = "'" + value;
-            }
-            fixedAnswers.put(item.getKey(), value);
+    public static String getFormattingResistantAnswer(TreeElement childElement) {
+        String answer = childElement.getValue() != null ? childElement.getValue().getDisplayText() : "";
+
+        if (!answer.isEmpty() && (childElement.getDataType() == Constants.DATATYPE_TEXT
+                || childElement.getDataType() == Constants.DATATYPE_MULTIPLE_ITEMS
+                || childElement.getDataType() == Constants.DATATYPE_BARCODE)) {
+            answer = "'" + answer;
```

Comment:
Oh, hah, I forgot values were made raw by manually adding the `'` and I thought it was a different Google Sheets API call. That's why I wasn't sure we could selectively make only certain types submit as raw.